samza-kafka/src/main/java/org/apache/samza/system/kafka/SamzaKafkaSystemAdmin.java

```diff
@@ -0,0 +1,667 @@
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.samza.Partition;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.KafkaConfig;
+import org.apache.samza.config.KafkaConsumerConfig;
+import org.apache.samza.config.SystemConfig;
+import org.apache.samza.system.ExtendedSystemAdmin;
+import org.apache.samza.system.StreamSpec;
+import org.apache.samza.system.StreamValidationException;
+import org.apache.samza.system.SystemStreamMetadata;
+import org.apache.samza.system.SystemStreamPartition;
+import org.apache.samza.util.ExponentialSleepStrategy;
+import org.apache.samza.util.ScalaJavaUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Function0;
+import scala.Function1;
+import scala.Function2;
+import scala.Option;
+import scala.collection.JavaConverters;
+import scala.runtime.AbstractFunction0;
+import scala.runtime.AbstractFunction1;
+import scala.runtime.AbstractFunction2;
+import scala.runtime.BoxedUnit;
+
+import static org.apache.samza.config.KafkaConsumerConfig.*;
+
+
+public class SamzaKafkaSystemAdmin<K, V> implements ExtendedSystemAdmin {
+  private static final Logger LOG = LoggerFactory.getLogger(SamzaKafkaSystemAdmin.class);
+
+  // The same default exponential sleep strategy values as in open source
+  private static final double DEFAULT_EXPONENTIAL_SLEEP_BACK_OFF_MULTIPLIER = 2.0;
+  private static final long DEFAULT_EXPONENTIAL_SLEEP_INITIAL_DELAY_MS = 500;
+  private static final long DEFAULT_EXPONENTIAL_SLEEP_MAX_DELAY_MS = 10000;
+  private static final int MAX_RETRIES_ON_EXCEPTION = 5;
+
+  private final String systemName;
+  private final Consumer<K, V> metadataConsumer;
+
+  private final Supplier<ZkUtils> connectZk;
+  private final Supplier<AdminClient> connectAdminClient;
+
+  // Custom properties to create a new coordinator stream.
+  private final Properties coordinatorStreamProperties;
+
+  // Replication factor for a new coordinator stream.
+  private final int coordinatorStreamReplicationFactor;
+
+  // Replication factor and kafka properties for changelog topic creation
+  private final Map<String, ChangelogInfo> changelogTopicMetaInformation;
+
+  // Kafka properties for intermediate topics creation
+  private final Map<String, Properties> intermediateStreamProperties;
+
+  // admin utils for some kafka related operations
+  // TODO refactor them out after we move to the new(broker based) AdminClient
+  private final KafkaSystemAdminUtilsScala kafkaAdminUtils;
+
+  // adminClient is required for deleteBeforeMessages operation
+  private final AdminClient adminClient;
+
+  private final boolean deleteCommittedMessages;
+
+  // used in TestRepartitionJoinWindowApp
+  @VisibleForTesting
+  public static volatile boolean deleteMessageCalled = false;
```

Comment:
Looks like this static constant is useful only for testing and not relevant with respect to the functionality of the API in itself.  Is it possible to replace this with a mock ?